### DISCUSSION TITLE
Current and Recent Options

### DISCUSSION THREAD

**Discussion by SouthpawSteven** [tidal-music/discussions#10] [Ideas & Feature requests]
Is it possible to open the API up so we can retrieve the users currently playing song, recently played tracks, and top/most played tracks? Spotify offers this so it would be a nice option for Tidal as well.

**Comment by stefanmoro** [comment]
Hi @SouthpawSteven and thanks for the suggestion!

This is functionality that we have briefly discussed internally and that we believe would be interesting to explore, so glad to hear there is interest for it.
Since we are just getting started with opening up our platform, we cannot promise anything at this stage, but all feedback is valuable for us when deciding what to focus on next.

We'll make sure to share any updates in the open!

Thanks

**Comment by keybrdist** [comment]
Just to "bump" this,  I wanted to also say we would definitely use this to notify our customers of their stats on Tidal.  This feature is essential for label owners and artists to keep them informed of their content performance.  

Another thing we would like to see is a way to search featured (editorial) content and chart placements. The more info we can pass to customers the better.

**Comment by lennartkloock** [comment]
For anyone that needs this endpoint for a hobby project (like me): Connect your account to your Last.fm account and then scrape this endpoint: https://www.last.fm/user/your_last_fm_username/partial/recenttracks?ajax=1
Replace `your_last_fm_username` with your Last.fm username
Your listening activity must be public on Last.fm for this to work without authorization. (See privacy settings)

Edit: No need to scrape the website, Last.fm also has an API endpoint for this: https://www.last.fm/api/show/user.getRecentTracks

**Comment by SubtleCo** [comment]
Hello! @stefanmoro I'm also here to bump this with a vote. My company is working with other DSPs to build a peripheral device that enhances a listener's experience. If Tidal were to open up a user based "now playing" route, we'd be able to include Tidal in our list of supported streaming providers. We'd love that! Can you tell me if there have been any more discussions around this?

Thank you!

**Comment by purpl3F0x** [comment]
Current playing functionality would be very welcomed, for my Discord RPC integration also

**Comment by eeitobi** [comment]
Was also searching for this on the API reference, currently using this feature on the Spotify API to feed a database with my personal listening habits that I can then visualize via a Grafana dashboard. 
Since I'm considering switching from Spotify to Tidal I was hoping to find something like a personal listening history on the API here also.

**Comment by Shelliis** [comment]
I recently switched from Spotiy to Tidal, but one thing that made me hesitant was the lack of an API to query my listened to songs. 
In the end, the better sound quality was decisive. But it was close and I really miss my personal stats.
Please consider to add such API's.

**Comment by probablyzal** [comment]
Did the switch from Spotify to TIDAL, best decision i've made. Need the API for a possible Discord integration!

**Reply by lennartkloock** [reply]
> Did the switch from Spotify to TIDAL, best decision i've made. Need the API for a possible Discord integration!

You can use the open-source desktop app [Tidal Hi-Fi](https://github.com/Mastermindzh/tidal-hifi). It has a built-in discord integration.

**Reply by probablyzal** [reply]
Thank you for this, i'll check it out!

**Reply by ungive** [reply]
> Need the API for a possible Discord integration!

I've recently released my own Discord integration:
https://github.com/ungive/discord-music-presence

Reddit post about it on the TIDAL subreddit:
https://www.reddit.com/r/TIdaL/comments/1d6azbm/i_made_a_discord_status_for_tidal_music_presence

While it's not integrated into TIDAL itself, like with Tidal Hi-Fi, it works with any media player and has quite a few customization options!

**Comment by stevefrenzel** [comment]
> Is it possible to open the API up so we can retrieve the users currently playing song, recently played tracks, and top/most played tracks? Spotify offers this so it would be a nice option for Tidal as well.

@stefanmoro any update on this? I would love to use any of these features and implement them on my website to give it a more personal touch. Come on, it would be such a cool feature! 🤗 I would also share in a blog post how I implemented it with Astro.

It's available in Spotify:

- [Get Currently Playing Track](https://developer.spotify.com/documentation/web-api/reference/get-the-users-currently-playing-track)
- [Get Recently Played Tracks](https://developer.spotify.com/documentation/web-api/reference/get-recently-played)
- [Get User's Top Items](https://developer.spotify.com/documentation/web-api/reference/get-users-top-artists-and-tracks)

**Reply by pkardas** [reply]
@stefanmoro we are very interested in this feature too. Are there any plans to expose it soon? 🙂

**Reply by devx9999** [reply]
It has been exposed.

**Reply by pkardas** [reply]
@tidal-music-star what do you mean? Has there been any update to the API recently that exposes recently played tracks?

**Comment by stefanmoro** [comment]
Hey everyone,

This functionality is unfortunately currently not planned. We see that there is big interest in it, and we'll get back as soon as we have any updates.

**Reply by RebellionBritney** [reply]
One of the reasons Spotify is so popular between people is that their API allows this, there are so many interesting apps/webs that use it to expand the experience between the music and the user. 
I think this should be your number 1 in your feature planned list. Hope we get news sooner than later

**Reply by pkardas** [reply]
This ☝️

**Reply by stevefrenzel** [reply]
@stefanmoro still interested in this feature 🤗

**Reply by ungive** [reply]
Quoting @RebellionBritney (emphasis mine):

> One of the reasons Spotify is so popular between people is that their API *[can do more than the TIDAL API]*

Spotify's popularity amongst developers might have just taken a big hit: https://community.spotify.com/t5/Spotify-for-Developers/Changes-to-Web-API/td-p/6540414. They removed several endpoints from their API which leaves many developers upset, understandably so. Maybe a TIDAL employee reading this will find this to be useful information! @stefanmoro

**Reply by pkardas** [reply]
@stefanmoro Just to note this here: we keep getting asked for this, so the demand is there :)

https://musicat.fm/notes/82512a96-090f-43e1-bb55-009595d575d1

**Reply by calgod** [reply]
Lots of demand for this from Airbuds Widget users: 
https://app.gleap.io/sharedboard/t6pW2tXQC5Z5bO13uAbo4rbdzsQlEfVu/featurerequests/7RiGYxbR032q9R6mCkk35kTmjwhpGLXVmz8CpABlN1gzsrXFx10C5c8kRwyDt8CWD14Wcf

It looks like Spotify, Apple Music, and SoundCloud all have APIs or SDKs that allow for querying a user's current/recent play history.

**Reply by FoxxMD** [reply]
Would also be interested in this! I'm the developer for [**Multi-Scrobbler**](https://foxxmd.github.io/multi-scrobbler/), an open source aggregator for scrobbling from many services. I've had Tidal requested many times but can't integrate Tidal without this functionality.

**Reply by benschwarz** [reply]
@stefanmoro, I have a new app and would integrate Tidal for user liked tracks & recent listening history. Please include me as a beta user 🤠

**Comment by xiandivyne** [comment]
I would like to bump this thread as to add further interest. I'm a streamer who swapped over from Spotify and would like to have a "Now Playing" overlay for OBS while streaming. Still lots of demand for this.

**Comment by liamlacey** [comment]
+1 for another person who has recently switched over from Spotify to Tidal and would like to see a now-playing and playing history added to the API.

**Comment by Bqleine** [comment]
+1, it would be great to connect Tidal to Listenbrainz!

**Comment by matrixfurry** [comment]
I would also really appreciate this feature. It would be very useful for connecting Tidal to platforms other than Last.fm, such as [Teal.fm](https://teal.fm), [Rocksky](https://rocksky.app), and [ListenBrainz](https://listenbrainz.com/).

**Reply by FoxxMD** [reply]
Since [this comment suggests](https://github.com/orgs/tidal-music/discussions/10#discussioncomment-7397705) that Tidal can be connected to Last.fm, you could do that and then use [Last.fm as a Source](https://foxxmd.github.io/multi-scrobbler/configuration/sources/lastfm-source/) in [multi-scrobbler](https://foxxmd.github.io/multi-scrobbler/) in order to mirror your last.fm scrobbles to the other [Clients](https://foxxmd.github.io/multi-scrobbler/configuration/clients/) you mentioned.

**Comment by PhilipTrauner** [comment]
+1! working on a service that allows users to explore their play history, which currently support Apple Music and Spotify. 
would love to add Tidal support!

**Comment by MMagneson** [comment]
I would also like this feature just to throw my voice into the chanting masses. With DJ'ing being well supported by Tidal, I want my VJ ecosystem to also be able to get the current track info like BPM from the DJ's currently playing tracks for instance. 

For event usage, this would also enable auto-tracking and auto filling reports to national rights-management organizations if you need something to convince the money to implement it!

**Comment by pojntfx** [comment]
I'm considering switching to Tidal, but the lack of this API has been keeping me from doing so because my personal website (https://felicitas.pojtinger.com/) has a little "I'm currently listening to <x>" widget that won't work without this API. Using last.fm would be nice, but my client of choice (Android) doesn't support it.

**Comment by kravetsone** [comment]
uhh this feature is really needed

**Comment by Jones-S** [comment]
+1
I would really appreciate to be able to fetch a users recently played tracks to get some insights into the history! This missing feature is currently keeping from making the switch.

**Comment by throny** [comment]
+1
I really want to scrobble to somewhere else but last.fm so opening up the API would be awesome.
